### PR TITLE
HPCC-24039 Default values for maxActive are not realistic

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [x] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [x] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday

**Comment by ghalliday** [comment]
Changes look ok.

The default for maxActive in eclccserver.cpp looks a bit strange:

            unsigned maxThreads = globals->getPropInt("@maxActive", useChildProcesses ? 100 : 4);

I suspect 4 would be more sensible for both situations.

File: dockerfiles/hpcc/values.yaml [hunk 1]
```diff
@@ -63,7 +63,7 @@ eclagent:
   ## replicas indicates how many eclagent pods should be started
   replicas: 1
   ## maxActive controls how many workunits may be active at once (per replica)
-  maxActive: 100
```
Comment (ghalliday) [inline dockerfiles/hpcc/values.yaml:66]: Also change default in agentexec.cpp?

**Review by ghalliday — COMMENTED** [review]
@richardkchapman a couple of comments

File: dockerfiles/hpcc/values.yaml (hunk 1 above)
Comment (richardkchapman) [reply dockerfiles/hpcc/values.yaml:66]: I was in two minds about this.

There's a difference between what we want the default to be if you don't specify in your helm chart, and what you want the default to be in values.yaml (representing a minimal system expected to be run on dodker desktop or similar).

If you are running on a full cloud and want to scale up according to load, 100 may be fine.

There's also an argument that we should be setting a similar limit for the child process case (in agentexec) as the k8s job case. Does setting childProcess imply you expect less parallelism, or that you expect smaller jobs?

**Comment by richardkchapman** [comment]
@ghalliday This seems to be stalled

**Comment by ghalliday** [comment]
I can merge this as is, because it is an improvement, but I think the defaults in the code also need fixing.  E.g. 100 threads in eclccserver when using child processes will bring any machine to its knees.  
100 in eclagent would possibly be ok if they didn't really do anything, but still seems too high.

**Comment by richardkchapman** [comment]
In non-containerized system, agentexec always operates in child process mode and does not set any limit. I think we therefore picked a very high limit to avoid breaking usage that relied on similar patterns.

**Comment by richardkchapman** [comment]
I don't mind setting eclccserver to 4 - but I am still unsure about correct default for agentexec

**Pull request merged**